### PR TITLE
(#18) Add configuration flag to always overwrite the cache

### PR DESCRIPTION
**Pull request by vjanelle**
* Add configuration flag to filesec/puppetsec to tell the provider to
always overwrite the local cache of a certificate with the one one
that's been presented (defaults to false for existing environments)
* Implement tests around configuration flag
* Update test certificates to replicate an updated certificate being
seen later

### CODE REVIEW

**Comment by vjanelle** [comment]
More than likely need go-choria#476 to be merged.

File: filesec/file_security.go [hunk 1]
```diff
@@ -69,6 +69,9 @@ type Config struct {
 
 	// DisableTLSVerify disables TLS verify in HTTP clients etc
 	DisableTLSVerify bool
+
+	//Support always overwriting the local filesystem cache
```
Comment (ripienaar) [inline filesec/file_security.go]: space after `//` and it should start with the var name, so `// AlwaysOverwriteCache support always....`

File: filesec/file_security_test.go [hunk 2]
```diff
@@ -505,6 +507,48 @@ var _ = Describe("FileSSL", func() {
 			Expect(stat.Size()).To(Equal(int64(16)))
 		})
 
+		It("Should always overwrite files", func() {
```
Comment (ripienaar) [inline filesec/file_security_test.go]: `Should support always overwriting files` would be clearer

File: filesec/file_security_test.go [hunk 3]
```diff
@@ -505,6 +507,48 @@ var _ = Describe("FileSSL", func() {
...
+
+			err = prov.CachePublicData(spd, identity)
+			Expect(err).To(BeNil())
+
+			cpd, err := prov.CachedPublicData(identity)
+			Expect(err).ToNot(HaveOccurred())
+
+			res := bytes.Compare(spd, cpd)
+			Expect(res).To(BeZero())
+		})
```
Comment (ripienaar) [inline filesec/file_security_test.go]: hmm, did we need all these new files and test data, i mean you could simply take a cert, write it to a temp directory cache and record its mtime, then attempt to cache again - if mtime change the cache updated? anyway no biggie this works too - mtime probably is 1 second resolution which would be annoying in tests

Comment (vjanelle) [reply filesec/file_security_test.go]: @ripienaar I didn't keep the CA private keys around >.<  so had to regenerate them.

I figure the byte slices would be small enough that it wouldn't matter in a test.

File: filesec/file_security.go [hunk 4]
```diff
@@ -69,6 +69,9 @@ type Config struct {
 
 	// DisableTLSVerify disables TLS verify in HTTP clients etc
 	DisableTLSVerify bool
+
+	// Support always overwriting the local filesystem cache
```
Comment (ripienaar) [inline filesec/file_security.go]: `// AlwaysOverwriteCache supports always overwriting the local filesystem cache`

**Comment by ripienaar** [comment]
nice one!

**Pull request merged**